### ISSUE TITLE
Preferred name is forgotten/removed without any indication why

### ISSUE THREAD

**Issue by not-inept** [Submitty/Submitty#3273]
I just received a report from a current RPI user that their preferred name was removed. This user had set a preferred first name at the beginning of the semester and discovered this week that their name on Submitty was changed back to their legal name. The name did not violate the rules stated in the 'Specify Preferred Names' pop-up and is the name the user goes by in regular/day-to-day life and on paper/in-class submissions.

It's unclear if a professor changed it (though the pop-up does indicate that is possible), but I wouldn't expect that in this case. Is there a good way for a user to determine why their name was changed, or any recent actions taken that may have unintentionally cleared the field? 

(This particular user was also super excited to see this feature existed after noticing the pencil icon next to their name, it's awesome to see inclusive feature work like this!)

**Comment by bmcutler** [comment]
Hi Mason,
I'm so sorry this happened!  We don't currently have logging or history on that data.  We did have an issue where the daily data feed from the registrar was overwriting Submitty-specificed preferred  name.  We may still have an issue there.  If the student is comfortable contacting me privately, I would like to followup and see if there's something specific happening that we can debug.
Thanks
Barb

**Comment by bmcutler** [comment]
Offline we discussed a plan to add logging to changes of preferred first & last names, so if something like this happens in the future we'll know when and who changed the name.  @pbailie is going to work on this logging.  

If anyone else experiences something similar, please contact the sysadmin for the system so we can debug the circumstances for the change.  Thanks :)

**Comment by not-inept** [comment]
Hey Barb,
Thanks for the quick response initially and for going ahead with next steps, I apologize for not getting back sooner. It's nice to see a familiar username here though!

The student who brought this up to me is comfortable reaching out to you. I've linked them this thread and they let me know that they'll send an email your way shortly. I really appreciate the steps you're taking here already. Definitely let me know if there is anything else I can do to help!

- Mason

**Comment by mcglow2** [comment]
@bmcutler, I discussed this feature with @pbailie and suggested that the database transaction log might be a good audit point for these changes.  That allows for the tracking of any possible way the data is altered.  The general idea would be to create a list of table.field items to audit, and then process the transaction log, recording any SQL INSERT/UPDATE/DELETE commands that contain them to a log file.

**Comment by bmcutler** [comment]
We indeed did have a bug in how the 'master' database synced with the individual databases for each course, which caused us to overwrite information with default values.   

It is likely that other students experienced this bug and had their preferred name overwritten.  We are very sorry for that mixup.  We have installed the bugfix on our RPI production server, so it should not happen going forward.  

We are also working on the logging/history component, once that is complete we'll be able to investigate if something like this happens in the future.
